### PR TITLE
[api] Mark ActivityExtensions.RecordException obsolete

### PR DESCRIPTION
**Pull request by ysolomchenko**
Fixes https://github.com/open-telemetry/opentelemetry-dotnet/issues/5665

## Changes

Mark ActivityExtensions.RecordException obsolete

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: src/OpenTelemetry.Api/Trace/ActivityExtensions.cs [hunk 1]
```diff
@@ -110,6 +111,7 @@ public static void RecordException(this Activity? activity, Exception? ex)
     /// <remarks> The exception is recorded as per <a href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/exceptions.md">specification</a>.
     /// "exception.stacktrace" is represented using the value of <a href="https://learn.microsoft.com/dotnet/api/system.exception.tostring">Exception.ToString</a>.
     /// </remarks>
+    [Obsolete("Call Activity.AddException instead this method will be removed in a future version.")]
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Api/Trace/ActivityExtensions.cs:114]: Should we replace the implementation of this method with a call to `activity.AddException`?

Comment (ysolomchenko) [reply src/OpenTelemetry.Api/Trace/ActivityExtensions.cs:114]: It appears that `activity.AddException` is not available in version 8.0.0 and was only introduced starting from version 9.0.0-rc.1.24431.7. Should we consider upgrading our version to `System.Diagnostics.DiagnosticSource v9.0.0-rc.1.24431.7`, or would we prefer to wait for the official release of v9.0.0?

File: src/OpenTelemetry.Api/Trace/ActivityExtensions.cs [hunk 2]
```diff
@@ -97,6 +97,7 @@ public static Status GetStatus(this Activity? activity)
...
```
Comment (cijothomas) [inline src/OpenTelemetry.Api/Trace/ActivityExtensions.cs:100]: can we explicitly call out that the end result is same when using the suggested alternative?

File: src/OpenTelemetry.Api/Trace/ActivityExtensions.cs (hunk 1 above)
Comment (cijothomas) [reply src/OpenTelemetry.Api/Trace/ActivityExtensions.cs:114]: https://github.com/open-telemetry/opentelemetry-dotnet/issues/5846 will be addressed soon by @rajkumar-rangaraj , so we can wait for that PR to merge first. @ysolomchenko or do you prefer to merge this as-is, and change the implementation to use actiity.AddException as a following after the other changes are in?

Comment (ysolomchenko) [reply src/OpenTelemetry.Api/Trace/ActivityExtensions.cs:114]: I think it would be better to wait until #5846 is done before finishing that PR

Comment (CodeBlanch) [reply src/OpenTelemetry.Api/Trace/ActivityExtensions.cs:114]: Just merged it 👍

**Review by cijothomas — APPROVED** [review]

**Review by reyang — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

**Pull request merged**